tests/functional/conftest.py

```diff
@@ -0,0 +1,19 @@
+from multiprocessing import Lock
+
+import pytest
+
+_db_start_lock = Lock()
+_DB_CLUSTER_STARTED = False
+
+
+# Running this should prevent tests from needing to be retried because the Databricks cluster isn't available
+@pytest.fixture(scope="class", autouse=True)
+def start_databricks_cluster(project, request):
+    global _DB_CLUSTER_STARTED
+    profile_type = request.config.getoption("--profile")
+    with _db_start_lock:
```

Comment:
Since this fixture gets used for all tests, including non-Databricks tests, does this force tests to be run entirely in series?

Reply:
it shouldn't since the lock would be released almost instantly (if not databricks or if `_DB_CLUSTER_STARTED` is `True`)

Reply:
it should only block the first databricks test that runs and even then only if the cluster isn't yet available